[awslogs] Auto-detect region and set user-agent

samuelkarp opened the pull request:
This pull request enhances the `awslogs` logging driver such that it can automatically detect the correct Amazon CloudWatch Logs region when the Docker daemon is running on an Amazon EC2 instance.

samuelkarp committed:
Update vendored aws-sdk-go

Signed-off-by: Samuel Karp <skarp@amazon.com>

samuelkarp committed:
[awslogs] Set User-Agent for Amazon CloudWatch Logs

Signed-off-by: Samuel Karp <skarp@amazon.com>

samuelkarp committed:
[awslogs] Auto-detect region on EC2 instances

Signed-off-by: Samuel Karp <skarp@amazon.com>

samuelkarp commented:
Any update here?

thaJeztah commented:
Hi @samuelkarp apologies for the delay, it's been busy times, so reviews are taking a bit longer than we want.

ping @calavera I saw you reviewed the original PR, perhaps you can have a look at this?

calavera commented on daemon/logger/awslogs/cloudwatchlogs.go:
```diff
@@ -100,6 +110,52 @@ func New(ctx logger.Context) (logger.Logger, error) {
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"error": err,
+			}).Error("Could not get region from EC2 metadata, environment, or log option", err)
```
did you mean to use `Errorf` here? It's going to print an unusual message if we leave it like this.

calavera commented:
I minor comment about error handling, otherwise LGTM.

tiborvass commented:
LGTM

moxiegirl commented on docs/reference/logging/awslogs.md:
```diff
@@ -34,9 +34,10 @@ You can use the `--log-opt NAME=VALUE` flag to specify Amazon CloudWatch Logs lo
-You must specify a region for the `awslogs` logging driver. You can specify the
-region with either the `awslogs-region` log option or `AWS_REGION` environment
-variable:
+The `awslogs` logging driver must send your Docker logs to a specified region;
```
The phrasing is a bit odd.  Maybe;

The `awslogs` logging driver sends your Docker logs to a specific region. Use the `awslogs-region` log option or the `AWS_REGION` environment variable to
set the region value. By default, if your Docker daemon is running on an EC2 instance and no region is set, the driver uses the instance's region.

samuelkarp commented:
@calavera and @moxiegirl, thanks for the review!  I've rebased on master and addressed both your comments.  (I'm seeing some failures in Jenkins, but I believe those to be unrelated to my code.)

thaJeztah commented:
@samuelkarp thanks! docs LGTM

ping @moxiegirl

calavera commented:
Failures are unrelated, but I'll wait for @moxiegirl to give her :+1: before merging.

moxiegirl commented:
LGTM! Thank you @samuelkarp

samuelkarp commented:
Thanks!
